File: backend/schemas/review.py
from datetime import datetime

from pydantic import BaseModel, Field


class ReviewSubmit(BaseModel):
    word_id: int
    quality: int = Field(..., ge=0, le=5)
    reviewed_at: datetime | None = None


class ReviewStatsResponse(BaseModel):
    due_count: int
    reviewed_today: int
    total_words: int


class ReviewCardResponse(BaseModel):
    word_id: int
    text: str
    phonetic: str | None = None
    ease_factor: float
    interval_days: int
    repetitions: int
    next_review: datetime
    algorithm: str = "sm2"
    phase: str = "new"
    difficulty: float | None = None
    stability: float | None = None
    retrievability: float | None = None
    scheduled_days: int | None = None
    learning_step: int = 0
    learning_due_at: datetime | None = None

File: frontend/src/types/review.ts
export interface ReviewCard {
  word_id: number;
  text: string;
  phonetic: string | null;
  definitions: { pos: string; meaning_zh: string; canvas_image: string | null; ink_data: string | null }[];
  ease_factor: number;
  interval_days: number;
  repetitions: number;
  next_review: string;
  algorithm: string;
  phase: string;
  difficulty: number | null;
  stability: number | null;
  retrievability: number | null;
  scheduled_days: number | null;
  learning_step: number;
  learning_due_at: string | null;
}

export interface ReviewStats {
  due_count: number;
  reviewed_today: number;
  total_words: number;
}

export interface ReviewSession {
  items: ReviewCard[];
  total: number;
  stats: ReviewStats;
}

export interface ReviewSubmit {
  word_id: number;
  quality: number;
  reviewed_at?: string;
}
